zipline/assets/roll_finder.py

```diff
@@ -179,20 +179,23 @@ def _active_contract(self, oc, front, back, dt):
         front_contract = oc.sid_to_contract[front].contract
         back_contract = oc.sid_to_contract[back].contract
 
-        # If the front contract has reached its auto close date, the back
-        # contract must be the active one, so return it immediately. Similarly,
-        # in the rare case that the back contract has not even started yet,
-        # short circuit here and return the front contract.
-        if dt >= front_contract.auto_close_date:
-            return back
-        elif dt < back_contract.start_date:
-            return front
-
         tc = self.trading_calendar
         trading_day = tc.day
         prev = dt - trading_day
         get_value = self.session_reader.get_value
 
+        # If the front contract is past its auto close date it cannot be the
+        # active contract, so return the back contract. Similarly, if the back
+        # contract has not even started yet, just return the front contract.
+        # The reason for comparing the back contract's start date to 'prev'
+        # instead of to 'dt' is because we need to get each contract's volume
+        # on the previous day, so we need to make sure the back contract exists
+        # on 'prev' in order to call 'get_value' below.
+        if dt > front_contract.auto_close_date:
```

Comment:
@yankees714 Just FYI, I made this change as you pointed out to be `>` instead of `>=` (and Jamie confirmed), and updated the tests accordingly (basically just moving rolls up by a day where necessary).